### ISSUE TITLE
The backpropagation problem of depth map supervision

### ISSUE THREAD

**Issue by everythoughthelps** [maturk/dn-splatter#73]
I think that if you use the rendered depth as another optimization target besides photometric loss, then you must calculate the gradient from depth. How did you complete the backpropagation part, I can't find it.

**Comment by maturk** [comment]
Hi @everythoughthelps, thanks for the interest. The gradients and in general the entire backend of Gaussian splatting (forward and backward passes of projection and rasterization) is taken care of within the [gsplat](https://github.com/nerfstudio-project/gsplat) dependency. Nerfstudio, and DN-Splatter, are only front-end PyTorch wrappers around main functionalities of gsplat. For more details about gsplat, I wrote a short document outlining the main parts and functionality [here](https://arxiv.org/abs/2409.06765).

In a sense, this repo enhances the basic functionality of gsplat with depth sensor, monocular depth, and/or normal information for regularization.
